### PR TITLE
Dapr bits

### PR DESCRIPTION
**Pull request by NicoVermeir**
## Summary

Describe your changes here.
- Adds documentation for using Dapr in .NET Aspire

I have not been able to preview this when rendered, I've based the syntax, and the overall structure of the document on the Orleans page

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/frameworks/dapr.md](https://github.com/dotnet/docs-aspire/blob/87be32e24b2c9b4376e1e6cef6fa9d2018b393bd/docs/frameworks/dapr.md) | [Use Dapr with .NET Aspire](https://review.learn.microsoft.com/en-us/dotnet/aspire/frameworks/dapr?branch=pr-en-us-929) |

### CODE REVIEW

**Comment by NicoVermeir** [comment]
@dotnet-policy-service agree

File: docs/frameworks/dapr.md [hunk 1]
```diff
@@ -13,7 +13,12 @@ In this guide, you'll learn how to take advantage of Dapr's abstraction and .NET
 
 ## Prerequisites
 
-- TODO: Add prerequisites
+- .NET 8.0 SDK or later
+- .NET Aspire workload
+- Docker
```
Comment (IEvangelist) [inline docs/frameworks/dapr.md:18]:
```suggestion
[!INCLUDE [aspire-prereqs](../includes/aspire-prereqs.md)]

In addition to the prerequisites for .NET Aspire, you will need:
```

File: docs/frameworks/dapr.md [hunk 2]
```diff
@@ -37,3 +42,71 @@ dotnet add package Aspire.Hosting.Dapr --prerelease
...
+```
+
+### [PackageReference](#tab/package-reference)
+
+```xml
+<PackageReference Include="Dapr.AspNetCore"
+                  Version="[SelectVersion]" />
+```
+
+Once installed into an ASP.Net Core project, the SDK can be added to the service builder.
```
Comment (IEvangelist) [inline docs/frameworks/dapr.md:86]:
```suggestion
---

Once installed into an ASP.NET Core project, the SDK can be added to the service builder.
```

File: docs/frameworks/dapr.md [hunk 3]
```diff
@@ -37,3 +42,71 @@ dotnet add package Aspire.Hosting.Dapr --prerelease
...
+
+## Dapr and .NET Aspire
+
+At first sight Dapr and .NET Aspire may look like they have overlapping functionality, and they do. However, they both take a different approach. .NET Aspire is an opiniated approach on how to build distributed applications on a cloud platform. Dapr is a runtime that abstracts away the common complexities of the underlying cloud platform. It relies on sidecars to provide abstractions for things like configuration, secret management and messaging. The underlying technology can be easily switched out through configuration files, while your code does not need to change.
+
+.NET Aspire makes setting up and debugging Dapr applications easier by providing a straightforward API to configure Dapr sidecars, and by exposing the sidecars as resources in the dashboard.
+
+[.NET Aspire Dapr sample app] (https://learn.microsoft.com/en-us/samples/dotnet/aspire-samples/aspire-dapr/)
```
Comment (IEvangelist) [inline docs/frameworks/dapr.md:112]:
```suggestion
[.NET Aspire Dapr sample app](/samples/dotnet/aspire-samples/aspire-dapr/)

```

File: docs/frameworks/dapr.md [hunk 4]
```diff
@@ -37,3 +42,71 @@ dotnet add package Aspire.Hosting.Dapr --prerelease
...
+
+Dapr uses the Sidecar pattern to run alongside your application. The Dapr sidecar is a lightweight, portable, and stateless HTTP server that listens for incoming HTTP requests from your application. The sidecar is responsible for managing the lifecycle of your application, including service discovery, configuration, and secrets management. To add a sidecar to a .NET Aspire resource by using the `WithDaprSidecar(string appId)` method. The `appId` parameter is the unique identifier for the Dapr application.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="18-21"  highlight="21":::
+
+The WithDaprSidecar method offers overloads to configure your Dapr sidecar options like app ID and ports. In the following example the Dapr sidecar is configured with specific ports for GRPC, HTTP, Metrics and a specific App ID.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="6-16"  highlight="6-12,16":::
+
+Putting everything together, here is an example of an Aspire AppHost project which includes:
```
Comment (IEvangelist) [inline docs/frameworks/dapr.md:59]:
```suggestion
Putting everything together, here's an example of a .NET Aspire app host project which includes:

```

File: docs/frameworks/dapr.md [hunk 5]
```diff
@@ -37,3 +42,71 @@ dotnet add package Aspire.Hosting.Dapr --prerelease
...
+The Dapr resource is added to the .NET Aspire distributed application builder using the `AddDapr()` method.
+An overload of the `AddDapr()` method that accepts Dapr options is available. For most applications, the default options will suffice.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="4":::
+
+Dapr uses the Sidecar pattern to run alongside your application. The Dapr sidecar is a lightweight, portable, and stateless HTTP server that listens for incoming HTTP requests from your application. The sidecar is responsible for managing the lifecycle of your application, including service discovery, configuration, and secrets management. To add a sidecar to a .NET Aspire resource by using the `WithDaprSidecar(string appId)` method. The `appId` parameter is the unique identifier for the Dapr application.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="18-21"  highlight="21":::
+
+The WithDaprSidecar method offers overloads to configure your Dapr sidecar options like app ID and ports. In the following example the Dapr sidecar is configured with specific ports for GRPC, HTTP, Metrics and a specific App ID.
```
Comment (IEvangelist) [inline docs/frameworks/dapr.md:55]:
```suggestion
The `WithDaprSidecar` method offers overloads to configure your Dapr sidecar options like app ID and ports. In the following example, the Dapr sidecar is configured with specific ports for GRPC, HTTP, metrics and a specific App ID.
```

**Review by IEvangelist — APPROVED** [review]

File: docs/frameworks/dapr.md [hunk 6]
```diff
@@ -37,3 +42,74 @@ dotnet add package Aspire.Hosting.Dapr --prerelease
...
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs":::
+
+The .NET Aspire dashboard will shows the Dapr sidecar as a resource, with its status and logs.
+
+:::image type="content" source="media/aspire-dashboard-dapr-sidecar-resources.png" lightbox="media/aspire-dashboard-dapr-sidecar-resources.png" alt-text=".NET Aspire dashboard showing Dapr sidecar resources":::
+
+## Adding the Dapr SDK
+
+To use Dapr APIs from .NET Aspire resources you can use the [Dapr SDK for ASP.Net Core](https://www.nuget.org/packages/Dapr.AspNetCore/). The Dapr SDK provides a set of APIs to interact with Dapr sidecars.
```
Comment (IEvangelist) [inline docs/frameworks/dapr.md:72]:
```suggestion
To use Dapr APIs from .NET Aspire resources you can use the [Dapr SDK for ASP.NET Core](https://www.nuget.org/packages/Dapr.AspNetCore/). The Dapr SDK provides a set of APIs to interact with Dapr sidecars.
```

File: docs/frameworks/dapr.md [hunk 7]
```diff
@@ -13,7 +13,12 @@ In this guide, you'll learn how to take advantage of Dapr's abstraction and .NET
 
 ## Prerequisites
 
-- TODO: Add prerequisites
+[!INCLUDE [aspire-prereqs](../includes/aspire-prereqs.md)]
+
+In addition to the prerequisites for .NET Aspire, you will need:
+- Dapr version 1.13 or later
```
Comment (IEvangelist) [inline docs/frameworks/dapr.md:19]:
```suggestion

- Dapr version 1.13 or later
```

File: docs/frameworks/dapr.md [hunk 8]
```diff
@@ -37,3 +43,74 @@ dotnet add package Aspire.Hosting.Dapr --prerelease
...
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="4":::
+
+Dapr uses the Sidecar pattern to run alongside your application. The Dapr sidecar is a lightweight, portable, and stateless HTTP server that listens for incoming HTTP requests from your application. The sidecar is responsible for managing the lifecycle of your application, including service discovery, configuration, and secrets management. To add a sidecar to a .NET Aspire resource by using the `WithDaprSidecar(string appId)` method. The `appId` parameter is the unique identifier for the Dapr application.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="18-21"  highlight="21":::
+
+The `WithDaprSidecar` method offers overloads to configure your Dapr sidecar options like app ID and ports. In the following example, the Dapr sidecar is configured with specific ports for GRPC, HTTP, metrics and a specific App ID.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="6-16"  highlight="6-12,16":::
```
Comment (IEvangelist) [inline docs/frameworks/dapr.md:58]:
```suggestion
:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="6-16"  highlight="1-7,11":::
```

**Comment by IEvangelist** [comment]
Hi @NicoVermeir - this is great thus far, here's what the preview looks like:

review learn microsoft com_en-us_dotnet_aspire_frameworks_dapr_branch=pr-en-us-929 tabs=dotnet-cli

**Pull request merged**